azure-ts-msi-keyvault-rbac/index.ts

```diff
@@ -0,0 +1,170 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as azure from "@pulumi/azure";
+import { execSync} from "child_process";
+
+// Create a resource group
+const resourceGroup = new azure.core.ResourceGroup("resourceGroup", {
+    location: "West US",
+});
+
+// Create a storage account for Blobs
+const storageAccount = new azure.storage.Account("storage", {
+    resourceGroupName: resourceGroup.name,
+    accountReplicationType: "LRS",
+    accountTier: "Standard",
+});
+
+// The container to put our files into
+const storageContainer = new azure.storage.Container("files", {
+    resourceGroupName: resourceGroup.name,
+    storageAccountName: storageAccount.name,
+    containerAccessType: "private",
+});
+
+// Azure SQL Server that we want to access from the application
+const sqlServer = new azure.sql.SqlServer("sqlserver", {
+    resourceGroupName: resourceGroup.name,
+
+    administratorLogin: "manualadmin",
+    administratorLoginPassword: "WeWontUseTh1sPwd",
+    version: "12.0",
+});
+
+// Azure SQL Database that we want to access from the application
+const database = new azure.sql.Database("sqldb", {
+    resourceGroupName: resourceGroup.name,
+    serverName: sqlServer.name,
+    requestedServiceObjectiveName: "S0"
+});
+
+// The connection string that has no credentials in it: authertication will come through MSI
+const connectionString = pulumi.all([sqlServer.name, database.name]).apply(([server, db]) => 
+    `Server=tcp:${server}.database.windows.net;Database=${db};`);
+
+// A file in Blob Storage that we want to access from the application
+const textBlob = new azure.storage.Blob("text", {
+    resourceGroupName: resourceGroup.name,
+    storageAccountName: storageAccount.name,
+    storageContainerName: storageContainer.name,
+    type: "block",
+    source: "./README.md",
+});
+
+// A plan to host the App Service
+const appServicePlan = new azure.appservice.Plan("asp", {
+    resourceGroupName: resourceGroup.name,
+    kind: "App",
+    sku: {
+        tier: "Basic",
+        size: "B1",
+    },
+});
+
+// ASP.NET deployment package
+const blob = new azure.storage.ZipBlob("zip", {
+    resourceGroupName: resourceGroup.name,
+    storageAccountName: storageAccount.name,
+    storageContainerName: storageContainer.name,
+    type: "block",
+
+    content: new pulumi.asset.FileArchive("./webapp/bin/Debug/netcoreapp2.2/publish")
+});
+
+const clientConfig = pulumi.output(azure.core.getClientConfig({}));
+const tenantId = clientConfig.apply(c => c.tenantId);
+
+// Determine the ID of the current user or service principal
+const currentPrincipal = clientConfig.apply(c => 
```

Comment:
Does it make sense to create a new SP programmatically and use that instead?  Is there a need to use the actively running Principal?

Reply:
The actively running principal has to be granted permissions to Key Vault so that it can actually add and then remove secrets from the Key Vault. Otherwise, `pulumi up` operations will fail.